packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart

```diff
@@ -103,10 +103,16 @@ class PurchaseParam {
   /// Do not pass in a clear text, your developer ID, the user’s Apple ID, or the
   /// user's Google ID for this field.
   /// For example, you can use a one-way hash of the user’s account name on your server.
-  final String applicationUserName;
+  final String? applicationUserName;
 
-  /// The 'sandboxTesting' is only available on iOS, set it to `true` for testing in AppStore's sandbox environment. The default value is `false`.
+  /// @deprecated Use [simulatesAskToBuyInSandbox] instead.
+  ///
+  /// This parameter has no effect and will be removed shortly.
+  @deprecated
   final bool sandboxTesting;
+
+  /// Only available on iOS, set it to `true` to produce an "ask to buy" flow for this payment in the sandbox.
+  final bool simulatesAskToBuyInSandbox;
```

Comment:
CC @LHLL 

This is the rename I talked about.

Reply:
I think Apple's documentation is very opaque, do we want to make it specific here? Maybe something like "Setting it to `true` will cause a transaction to be in the state `SKPaymentTransactionStateDeferred` and produce an "ask to buy" prompt that interrupts the the payment flow."

Reply:
Done. I added the detailed description in `sk_payment_queue_wrapper.dart` and add a link here.